src/artery/utility/CMakeLists.txt

```diff
@@ -7,3 +7,9 @@ target_sources(core PUBLIC
     FilterRules.cc
     Geometry.cc
 )
+
+add_library(asio-data-message OBJECT)
+generate_opp_message(MESSAGE AsioData.msg TARGET asio-data-message)
+
+artery_mark_as_codegen(asio-data-message)
+target_link_libraries(core PRIVATE asio-data-message)
```

Comment:
Clumsy again. This should take at most two lines: generate_opp_message and target_link_libraries